Patch proposed upstream, but yet to be committed. Should be redundant after
next MPlayer update.
--- MPlayer/libavformat/avio.c  (revision 11309)
+++ MPlayer/libavformat/avio.c  (working copy)
@@ -110,9 +110,9 @@
     return ret;
 }
 
-#if defined(CONFIG_MUXERS) || defined(CONFIG_PROTOCOLS)
 int url_write(URLContext *h, unsigned char *buf, int size)
 {
+#if defined(CONFIG_MUXERS) || defined(CONFIG_PROTOCOLS)
     int ret;
     if (!(h->flags & (URL_WRONLY | URL_RDWR)))
         return AVERROR(EIO);
@@ -121,8 +121,10 @@
         return AVERROR(EIO);
     ret = h->prot->url_write(h, buf, size);
     return ret;
+#else
+    return AVERROR(EIO);
+#endif //CONFIG_MUXERS || CONFIG_PROTOCOLS
 }
-#endif //CONFIG_MUXERS || CONFIG_PROTOCOLS
 
 offset_t url_seek(URLContext *h, offset_t pos, int whence)
 {
